rotkehlchen/chain/ethereum/tokens.py

```diff
@@ -141,116 +156,59 @@ def query_tokens_for_addresses(
         for asset in ignored_assets:  # don't query for the ignored tokens
             if asset.is_eth_token():  # type ignore since we know asset is a token
                 exceptions.append(EthereumToken.from_asset(asset).ethereum_address)  # type: ignore
         all_tokens = GlobalDBHandler().get_ethereum_tokens(
             exceptions=exceptions,
             except_protocols=['balancer'],
         )
-        # With etherscan with chunks > 120, we get request uri too large
-        # so the limitation is not in the gas, but in the request uri length
-        etherscan_chunks = list(get_chunks(all_tokens, n=ETHERSCAN_MAX_TOKEN_CHUNK_LENGTH))
-        other_chunks = list(get_chunks(all_tokens, n=OTHER_MAX_TOKEN_CHUNK_LENGTH))
-        now = ts_now()
-        token_usd_price: Dict[EthereumToken, Price] = {}
-        result = {}
-
-        for address in addresses:
-            with self.db.user_write() as cursor:
-                saved_list = self.db.get_tokens_for_address_if_time(cursor, address=address, current_time=now)  # noqa: E501
-                if force_detection or saved_list is None:
-                    balances = self.detect_tokens_for_address(
-                        write_cursor=cursor,
-                        address=address,
-                        token_usd_price=token_usd_price,
-                        etherscan_chunks=etherscan_chunks,
-                        other_chunks=other_chunks,
-                    )
-                else:
-                    if len(saved_list) == 0:
-                        continue  # Do not query if we know the address has no tokens
-
-                    balances = defaultdict(FVal)
-                    self._get_tokens_balance_and_price(
-                        address=address,
-                        tokens=saved_list,
-                        balances=balances,
-                        token_usd_price=token_usd_price,
-                        call_order=None,  # use defaults
-                    )
-
-            result[address] = balances
-
-        return result, token_usd_price
-
-    def _get_tokens_balance_and_price(
-            self,
-            address: ChecksumEthAddress,
-            tokens: List[EthereumToken],
-            balances: Dict[EthereumToken, FVal],
-            token_usd_price: Dict[EthereumToken, Price],
-            call_order: Optional[Sequence[WeightedNode]],
-    ) -> None:
-        ret = self._get_multitoken_account_balance(
-            tokens=tokens,
-            account=address,
-            call_order=call_order,
-        )
-        for token_identifier, value in ret.items():
-            token = EthereumToken.from_identifier(token_identifier)
-            if token is None:  # should not happen
-                log.warning(
-                    f'Could not initialize token with identifier {token_identifier}. '
-                    f'Should not happen. Skipping its token balance query',
+        for account in accounts:
```

Comment:
Change it up a bit here.


Since only `n` and `querying_etherscan` is what is differentiated here simply set the variables appropriately depending on `self.ethereum.connected_to_any_web3()` and then after the conditional call `self._query_chunks` with the customized variables.

Reply:
Done